## *********************************************************
##
## File autogenerated for the parallel_planner package
## by the dynamic_reconfigure package.
## Please do not edit.
##
## ********************************************************/

from dynamic_reconfigure.encoding import extract_params

inf = float('inf')

config_description = {'name': 'Default', 'type': '', 'state': True, 'cstate': 'true', 'id': 0, 'parent': 0, 'parameters': [{'name': 'kf_update_rate', 'type': 'double', 'default': 20.0, 'level': 0, 'description': 'Частота обновления фильтра Калмана (Гц)', 'min': 1.0, 'max': 100.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'kf_pos_noise', 'type': 'double', 'default': 0.1, 'level': 0, 'description': 'Ковариация шума по положению', 'min': 0.0, 'max': 10.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'kf_vel_noise', 'type': 'double', 'default': 0.05, 'level': 0, 'description': 'Ковариация шума по скорости', 'min': 0.0, 'max': 10.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'dsm_max_accel', 'type': 'double', 'default': 2.0, 'level': 0, 'description': 'Максимальное ускорение (м/с^2)', 'min': 0.1, 'max': 10.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'dsm_max_jerk', 'type': 'double', 'default': 5.0, 'level': 0, 'description': 'Максимальный рывок (м/с^3)', 'min': 0.1, 'max': 20.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'vfh_horizontal_res', 'type': 'int', 'default': 5, 'level': 0, 'description': 'Горизонтальная дискретизация (градусы)', 'min': 1, 'max': 45, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'int', 'cconsttype': 'const int'}, {'name': 'vfh_vertical_res', 'type': 'int', 'default': 5, 'level': 0, 'description': 'Вертикальная дискретизация (градусы)', 'min': 1, 'max': 45, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'int', 'cconsttype': 'const int'}, {'name': 'vfh_clearance_radius', 'type': 'double', 'default': 0.8, 'level': 0, 'description': 'Минимальное безопасное расстояние (м)', 'min': 0.1, 'max': 5.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'vfh_cost_yaw_weight', 'type': 'double', 'default': 1.0, 'level': 0, 'description': 'Вес предпочтения по курсу', 'min': 0.0, 'max': 10.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'vfh_cost_goal_weight', 'type': 'double', 'default': 3.0, 'level': 0, 'description': 'Вес предпочтения по направлению к цели', 'min': 0.0, 'max': 10.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'vfh_obstacle_threshold', 'type': 'int', 'default': 5, 'level': 0, 'description': 'Порог засоренности для препятствия', 'min': 0, 'max': 100, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'int', 'cconsttype': 'const int'}, {'name': 'sensor_range_max', 'type': 'double', 'default': 10.0, 'level': 0, 'description': 'Максимальная дальность сенсора (м)', 'min': 1.0, 'max': 50.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'max_forward_speed', 'type': 'double', 'default': 2.0, 'level': 0, 'description': 'Максимальная поступательная скорость (м/с)', 'min': 0.0, 'max': 10.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'goal_x', 'type': 'double', 'default': 15.0, 'level': 0, 'description': 'Цель X', 'min': -100.0, 'max': 100.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'goal_y', 'type': 'double', 'default': 15.0, 'level': 0, 'description': 'Цель Y', 'min': -100.0, 'max': 100.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}, {'name': 'goal_z', 'type': 'double', 'default': 4.0, 'level': 0, 'description': 'Цель Z', 'min': 0.0, 'max': 20.0, 'srcline': 273, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'edit_method': '', 'ctype': 'double', 'cconsttype': 'const double'}], 'groups': [], 'srcline': 245, 'srcfile': '/opt/ros/noetic/lib/python3/dist-packages/dynamic_reconfigure/parameter_generator.py', 'class': 'DEFAULT', 'parentclass': '', 'parentname': 'Default', 'field': 'default', 'upper': 'DEFAULT', 'lower': 'groups'}

min = {}
max = {}
defaults = {}
level = {}
type = {}
all_level = 0

#def extract_params(config):
#    params = []
#    params.extend(config['parameters'])
#    for group in config['groups']:
#        params.extend(extract_params(group))
#    return params

for param in extract_params(config_description):
    min[param['name']] = param['min']
    max[param['name']] = param['max']
    defaults[param['name']] = param['default']
    level[param['name']] = param['level']
    type[param['name']] = param['type']
    all_level = all_level | param['level']

